src/ng/urlUtils.js

```diff
@@ -94,3 +95,22 @@ function urlIsSameOrigin(requestUrl) {
   return (parsed.protocol === originUrl.protocol &&
           parsed.host === originUrl.host);
 }
+
+/**
+ * Parse a request URL and determine whether it is same-origin as the document base URL.
+ *
+ * Note: The base URL is usually the same the document location (`location.href`) but can
+ * overriden by using the `<base>` tag.
+ *
+ * @param {string|object} requestUrl The url of the request as a string that will be resolved
+ * or a parsed URL object.
+ * @returns {boolean} Whether the URL is same-origin as the document base URL.
+ */
+function urlIsSameOriginAsBaseUrl(requestUrl) {
```

Comment:
Nit: I would rather not duplicate the code. Maybe have a base function that `urlIsSameOrigin`/`urlIsSameOriginAsBase` can delegate two (and pass the `originUrl`/`baseUrl` as parameters).